catalyst-gateway/bin/src/cli.rs

```diff
@@ -48,8 +52,48 @@ impl Cli {
             Self::Run(settings) => {
                 logger::init(settings.log_level)?;
 
+                // Tick every N seconds until config exists in db
+                let check_config_tick = env::var("CHECK_CONFIG_TICK")?;
+
+                // Tick every N seconds until data is stale enough to update
+                let data_refresh_tick = env::var("DATA_REFRESH_TICK")?;
+
+                // Unique machine id
+                let machine_id = env::var("MACHINE_ID")?;
+
                 let state = Arc::new(State::new(Some(settings.database_url)).await?);
-                service::run(&settings.docs_settings, state).await?;
+                let event_db = state.event_db()?;
+
+                tokio::spawn(async move {
+                    match service::run(&settings.docs_settings, state.clone()).await {
+                        Ok(()) => info!("Endpoints started ok"),
+                        Err(err) => {
+                            error!("Error starting endpoints {err}");
+                        },
+                    }
+                });
+
+                // tick until config exists
+                let mut interval =
+                    time::interval(time::Duration::from_secs(check_config_tick.parse::<u64>()?));
+                let config = loop {
```

Comment:
Looks for me it makes sense to move this code under the `start_follower`.
WDYT ?

Reply:
We can't start the follower until there is a config in the DB.